projects/igniteui-angular/src/lib/progressbar/linearbar.component.spec.ts

```diff
@@ -45,405 +44,259 @@ describe('IgxLinearBar', () => {
 
-        tick(tickTime);
-        fixture.detectChanges();
-
         expect(progress.value).toBe(progressBarValue);
 
         progressBarMaxValue = 100;
         progress.max = progressBarMaxValue;
 
-        tick(tickTime);
-        fixture.detectChanges();
-
         expect(progress.value).toBe(progressBarValue);
-    }));
+    });
 
     it('should not update value if max is increased (without animation)', () => {
         let progressBarMaxValue = 100;
         const progressBarValue = 50;
-        const fixture = TestBed.createComponent(InitLinearProgressBarComponent);
-
-        fixture.detectChanges();
 
-        const progress = fixture.componentInstance.linearBar;
-        progress.animate = false;
         progress.max = progressBarMaxValue;
         progress.value = progressBarValue;
 
-        fixture.detectChanges();
-
         expect(progress.value).toBe(progressBarValue);
 
         progressBarMaxValue = 200;
         progress.max = progressBarMaxValue;
 
-        fixture.detectChanges();
-
         expect(progress.value).toBe(progressBarValue);
     });
 
-    it('Should update value when we try to decrese it', () => {
-        const fixture = TestBed.createComponent(LinearBarComponent);
-        fixture.detectChanges();
-        const progressBar = fixture.componentInstance.progressbar;
-        let expectedValue = 50;
+    it('when update step is bigger than passed value the progress indicator should follow the value itself', () => {
+        const step = 5;
+        const value = 2;
+        const max = 10;
+        progress.step = step;
+        progress.max = max;
+        progress.value = value;
+
+        const percentValue = toPercent(value, max);
+        expect(progress.value).toBe(value);
+        expect(progress.step).toBe(step);
+        expect(progress.max).toBe(max);
+        expect(progress.valueInPercent).toBe(percentValue);
+
+        // Should not set a step larger than max value
+        progress.step = 20;
+        expect(progress.step).toBe(step);
+        expect(progress.max).toBe(max);
+    });
 
-        fixture.componentInstance.value = expectedValue;
+    it('When indeterminate mode is on value should not be updated', () => {
+        expect(progress.value).toEqual(0);
 
-        fixture.detectChanges();
+        progress.indeterminate = true;
+        progress.value = 30;
+        expect(progress.value).toEqual(0);
 
-        expect(progressBar.value).toBe(expectedValue);
+        progress.value = 20;
+        expect(progress.value).toEqual(0);
+    });
 
-        expectedValue = 20;
-        fixture.componentInstance.value = expectedValue;
+    it('Prevent constant update of progress value when value and max value differ', () => {
+        const maxVal = 3.25;
+        const value = 2.55;
+        const progressBar = fixture.debugElement.nativeElement;
+
+        expect(progressBar.attributes['aria-valuenow'].textContent).toBe(progress.value.toString());
+        progress.step = 0.634;
```

Comment:
Passing this value doesn't affect the test. It will always be valid no matter the step.

Reply:
I have not changed this test from the previous batch.